src/hotspot/share/opto/callGenerator.cpp

```diff
@@ -728,7 +800,7 @@ class LateInlineBoxingCallGenerator : public LateInlineCallGenerator {
 
  public:
   LateInlineBoxingCallGenerator(ciMethod* method, CallGenerator* inline_cg) :
-    LateInlineCallGenerator(method, inline_cg, /*is_pure=*/true) {}
+    LateInlineCallGenerator(method, inline_cg, /*is_pure=*/true, /*is_boxing*/true) {}
```

Comment:
You can replace `_is_boxing_call` with `is_boxing_late_inline()`. I find it clearer.

```
src/hotspot/share/opto/callGenerator.cpp
@@ -813,6 +795,8 @@ class LateInlineBoxingCallGenerator : public LateInlineCallGenerator {
     return new_jvms;
   }
 
+  virtual bool is_boxing_late_inline() const { return true; }
+
   virtual CallGenerator* with_call_node(CallNode* call) {
     LateInlineBoxingCallGenerator* cg = new LateInlineBoxingCallGenerator(method(), _inline_cg);
     cg->set_call_node(call->as_CallStaticJava());

src/hotspot/share/opto/callGenerator.hpp
@@ -75,6 +75,7 @@ class CallGenerator : public ResourceObj {
   virtual bool      is_late_inline() const         { return false; }
   // same but for method handle calls
   virtual bool      is_mh_late_inline() const      { return false; }
+  virtual bool      is_boxing_late_inline() const  { return false; }
   virtual bool      is_string_late_inline() const  { return false; }
   virtual bool      is_virtual_late_inline() const { return false; }
```

Reply:
Thank you for your review. I fix it in my patch.